File: examples/url_utf8.pl
use v5.14;
use utf8;

use Mojo::URL;
use Encode;

my $url1 = Mojo::URL->new('http://pl.wikipedia.org')->path('/w/index.php')->query([title=>'Wikipedia:Strona_główna',action=>'history']);
say $url1;

my $url2 = Mojo::URL->new('http://pl.wikipedia.org')->path('/wiki/Wikipedia:Strona_główna');
say $url2;

my $url3 = Mojo::URL->new('http://żółwie.pl');
say $url3;

my $url4 = Mojo::URL->new('http://localhost')->userinfo(encode_utf8('gęślą:jaźń'));
say $url4;

File: examples/url_utf8.py
# -*- coding: utf-8 -*-

from __future__ import print_function

import Pyjo.URL

url1 = Pyjo.URL.new('http://pl.wikipedia.org').set(path='/w/index.php', query=[('title', u'Wikipedia:Strona_główna'), ('action', 'history')])
print(url1)

url2 = Pyjo.URL.new('http://pl.wikipedia.org').set(path=u'/wiki/Wikipedia:Strona_główna')
print(url2)

url3 = Pyjo.URL.new(u'http://żółwie.pl')
print(url3)

url4 = Pyjo.URL.new(u'http://localhost').set(userinfo=u'gęślą:jaźń')
print(url4)
